### PR TITLE
Add support for renaming actions, conditions and triggers

### PR DESCRIPTION
**Pull request by frenck**
## Proposed change

Adds support for renaming triggers, conditions, and actions

CleanShot 2022-08-23 at 00 22 23

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue or discussion:
- Link to documentation pull request:

## Checklist

- [x] The code change is tested and works locally.
- [x] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

File: src/panels/config/automation/condition/ha-automation-condition-row.ts [hunk 1]
```diff
@@ -285,6 +294,31 @@ export default class HaAutomationConditionRow extends LitElement {
     }
   }
 
+  private async _renameCondition(): Promise<void> {
+    const describedCondition = describeCondition(this.condition);
+    const alias = await showPromptDialog(this, {
```
Comment (balloob) [inline src/panels/config/automation/condition/ha-automation-condition-row.ts:299]: Can we fix the prompt dialog? The input element should be a lot wider. It’s too small.

File: src/panels/config/automation/condition/ha-automation-condition-row.ts [hunk 2]
```diff
@@ -285,6 +294,31 @@ export default class HaAutomationConditionRow extends LitElement {
...
+      title: this.hass.localize(
+        "ui.panel.config.automation.editor.conditions.change_alias"
+      ),
+      inputLabel: this.hass.localize(
+        "ui.panel.config.automation.editor.conditions.alias"
+      ),
+      inputType: "string",
+      defaultValue: this.condition.alias || describedCondition,
```
Comment (balloob) [inline src/panels/config/automation/condition/ha-automation-condition-row.ts:307]: The `describedCondition`, can we put that as a placeholder? We wouldn’t want people to open rename, then hit ok without doing anything and then ending up with an alias set to the current description.

File: src/panels/config/automation/condition/ha-automation-condition-row.ts [hunk 3]
```diff
@@ -294,6 +328,7 @@ export default class HaAutomationConditionRow extends LitElement {
   static get styles(): CSSResultGroup {
     return [
       haStyle,
+      haStyleDialog,
```
Comment (balloob) [inline src/panels/config/automation/condition/ha-automation-condition-row.ts:331]: Why is this necessary?

Comment (frenck) [reply src/panels/config/automation/condition/ha-automation-condition-row.ts:331]: uh, it's not indeed. Thanks 👍

File: src/panels/config/automation/condition/ha-automation-condition-row.ts (hunk 2 above)
Comment (frenck) [reply src/panels/config/automation/condition/ha-automation-condition-row.ts:307]: > The describedCondition, can we put that as a placeholder?

I don't believe the prompt dialog supports that, will take a look (or else add it).

>  then hit ok without doing anything and then ending up with an alias set to the current description.

That case is handled a couple of lines below.

Comment (frenck) [reply src/panels/config/automation/condition/ha-automation-condition-row.ts]: Added support for placeholders.

File: src/panels/config/automation/condition/ha-automation-condition-row.ts (hunk 1 above)
Comment (frenck) [reply src/panels/config/automation/condition/ha-automation-condition-row.ts:299]: Fixed the prompt dialog, and made it a little wider.

**Comment by zsarnett** [comment]
Looks good. But has merge conflicts now 😬

**Comment by frenck** [comment]
Expected :) Will fix in a sec.

**Comment by frenck** [comment]
Rebased to deal with the merge conflict

**Review by zsarnett — APPROVED** [review]
